flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -232,7 +255,7 @@ private void addValue(OUT reuse) throws IOException, SQLException {
 						reuse.setField(resultSet.getInt(pos + 1), pos);
 						break;
 					case java.sql.Types.FLOAT:
-						reuse.setField(resultSet.getDouble(pos + 1), pos);
+						reuse.setField(resultSet.getFloat(pos + 1), pos);
```

Comment:
The problem arises when I look at the column type in the db and I see Float, so I set Float as the Class of my tuple field and then I get a ClassCastException..